src/native/eventpipe/ds-ipc-pal-socket.c

```diff
@@ -356,9 +353,9 @@ ipc_socket_create_tcp (DiagnosticsIpc *ipc)
 	new_socket = socket (ipc->server_address_family, socket_type, IPPROTO_TCP);
 	if (new_socket != DS_IPC_INVALID_SOCKET) {
 #ifndef SOCK_CLOEXEC
-		if (fcntl (new_socket, F_SETFD, FD_CLOEXEC) == -1) {
-			EP_ASSERT (!"Failed to set CLOEXEC");
-		}
+#ifndef HOST_WIN32
```

Comment:
This should still check the existence of SOCK_CLOEXEC so we only do the fcntl call if we didn't already set it during the call to socket.

Reply:
It is already under `#ifndef SOCK_CLOEXEC`, or maybe you though about adding check for FD_CLOEXEC as well? That is not something we have done in other places doing similar code (just checking `SOCK_CLOEXEC`), and I know that some Mono headers actually define this on platforms where its not used/available, so on those platforms the additional check won't make any difference.

Reply:
Ah, no I missed line 355 because I was reviewing this on my phone 🙄. If we're checking `ifndef SOCK_CLOEXEC` this is fine.

nit: `#if !defined(SOCK_CLOEXEC) && !defined(HOST_WIN32)`?